### PR TITLE
Add multi-transaction functionality to Gnosis Safe Control dialog

### LINKED ISSUE #3563: Add "Add another transaction" functionality to Gnosis Control Safe Dialog
### Story

As a user, when I click on the "Add another transaction" button a new section, where I can create another transaction, should appear

### Description

The new section that appears when clicking the button is like the first one so that you can create a brand new, different, transaction. The transactions should now be inside tabs that can be opened, closed, or deleted.

### Design

[Figma link](https://www.figma.com/file/LlVmeMQxHVA04evfrgZyuN/Safe-Control?node-id=139%3A9825)

FireShot Capture 641 - Safe Control – Figma - www figma com
FireShot Capture 642 - Safe Control – Figma - www figma com
FireShot Capture 644 - Safe Control – Figma - www figma com

### This issue is complete when...

- [x] Clicking on the "Add another transaction" button adds another transaction section to the dialog.
- [x] Adding a transaction closes the tabs of the previous transaction and only leaves the new one open.
- [x] Hovering over the tab heading makes the "Delete transaction" icon appear and when clicking it, it makes the section go away.
- [x] Each tab heading has a title with the transaction number and type, and on click, it opens and closes the tab according to its current state.
- [x] The form validation is dynamic, it should adapt to having more or fewer fields to validate in order to continue.

### PR DESCRIPTION
**Pull request by ArmandoGraterol**
Added the possibility of creating multiple transactions to the safe control dialog. Each transaction should now be inside a tab that can be opened, closed, and/or deleted.

The validation of the form should also now be dynamic and adapt to the different numbers of transactions and disable the submit button if the validation fails in any of the transactions.

FireShot Capture 657 - Actions - Colony - wonker - localhost
FireShot Capture 658 - Actions - Colony - wonker - localhost

Resolves #3563

### CODE REVIEW

**Review by willm30 — CHANGES_REQUESTED** [review]
Carets could be aligned: 

align-carets

Validation message should be user-friendly:

err-msg
err-msg2

Is there anyway to stop my browser from thinking this is a residential address input? The autofill is overlaying the wallet addresses. 

form-val

When you have two transactions and you delete the first one, they both disappear and the form becomes un-interactive.  

multi-trans

When you have two of the same transaction and delete the second one, they both disappear, and then when you click "add another transaction", the first one reappears. 

double-trans

**Comment by ArmandoGraterol** [comment]
> Carets could be aligned:
> 
> align-carets

I've improved the alignment of the carets in the tabs relative to the carets in the select fields, as they are aligned in the specs, however the rest I'll leave as they are because changing them implies changing the SVG or core components, which is something that I already did in #3506 (Not on this feature branch) and it is quite a painful process :)

**Review by willm30 — APPROVED** [review]
Working as expected! Just made a note about wrapping the Form in an actual `<form>` tag, as I believe this is necessary for the `autocomplete='off'` to work on the individual inputs (or, at least, this is the behaviour I encountered in my local environment).

**Review by danbr — APPROVED** [review]
Looks good overall, just a couple of minor points.... to check/add - but I'll approve. Well done.

Should the confirm button copy display 'Interact' or 'Create transaction'? 
<NEWSFLASH> @chinins just told me that she already encountered this discrepancy, and as expected - you correctly used the appropriate option (well done you)

`Max` button missing from amount.......
FIgma
Screenshot 2022-07-12 at 16 48 06

File: src/modules/dashboard/components/Dialogs/GnosisControlSafeDialog/GnosisControlSafeForm.tsx [hunk 1]
```diff
@@ -116,6 +138,56 @@ const GnosisControlSafeForm = ({
...
+    },
+    [],
+  );
+  const autogeneratedIds = useMemo(
+    () => [...new Array(values.transactions.length)].map(nanoid),
+    [values.transactions.length],
+  );
 
   return (
     <>
```
[the remark this answers is not in the document]
Comment (ArmandoGraterol) [reply src/modules/dashboard/components/Dialogs/GnosisControlSafeDialog/GnosisControlSafeForm.tsx:193]: Hmmm, we do not support Brave, only Google Chrome. I know that Brave is based on Chrome but as we can see here, there are differences.

When you are testing or developing things, I would advise you to use Google Chrome as that's officially the only browser that Colony is developed in, and to.

**Comment by ArmandoGraterol** [comment]
> Should the confirm button copy display 'Interact' or 'Create transaction'?
> @chinins just told me that she already encountered this discrepancy, and as expected - you correctly used the appropriate option (well done you)

Ah, yes, just as I planned all along (??

> Max button missing from amount.......

I didn't include it the first time because I saw that the other amount fields for tokens that we have don't max buttons, so for consistency's sake, I left it like that, but yeah I can include it and if it doesn't make sense later then we can just remove it :)

File: src/modules/dashboard/components/Dialogs/GnosisControlSafeDialog/GnosisControlSafeForm.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (willm30) [reply src/modules/dashboard/components/Dialogs/GnosisControlSafeDialog/GnosisControlSafeForm.tsx:193]: Ok good to know

**Pull request merged**